standards/mobile_app_standards.md

```diff
@@ -0,0 +1,135 @@
+
+Unfortunately, the DEFRA family mandating of Apple devices does have some negative implications for Progressive Web Apps, of which we need to be aware. Apple force the use of the Safari/WebKit browser on all iOS/iPad OS devices, and, of all the major browsers, Safari has the most limited support for Progressive Web Apps. There is no way around this as even other browser apps in the Apple app store, such as Chrome, are merely wrappers for WebKit. The most notable restrictions are likely to be around size of storage (though, in January 2020, and if using IndexedDB, which is usually the most functional local storage option in any case, over 1 GB was found to be available on a DEFRA iPad), persistence of stored data (as of March 2020 you can depend on IndexedDB data being retained for at least 2 weeks, and possibly much longer, even if an app is completely unused, and data has been observed to survive an iPad OS and Safari update. This therefore seems to be a diminishing issue, though one of which you should be aware.) and the fact that, in Safari on iOS or iPad OS, the way you install an app on the home screen is still currently non-standard.
+
+It is possible to build a perfectly useful PWA that runs on Safari, including being able to access native features such as taking photos (though, at the time of writing, only stills and not videos) and accessing geolocation. However, the situation is always changing, usually improving, though that cannot be guaranteed, so you must use a site such as <https://whatwebcando.today/> to assess which features are actually available in Safari on a DEFRA Apple device before committing to delivering a PWA. Similarly, you should assess the current storage capabilities of Safari on a device by using a web site such as <https://demo.agektmr.com/storage/>.
+
+Of course, PWAs are storing data locally: otherwise they cannot work offline. That means that you will also need to consider the security of the data at rest on the device. These considerations are slightly different for PWAs than for other apps, as the data is being retained in the browser. If, as is likely and recommended, you are using IndexedDB to store your data, then all browsers compliant with the W3C spec, such as Safari, will only allow JavaScript from the same origin as the PWA that has created the database to access that data. Also, your data will be protected by your device's normal data encryption: for example, for an internally managed DEFRA iPad where the setting of a passcode is enforced, that means the data will be encrypted whenever the device is turned off, and only decrypted when the user re-enters the correct passcode. That means "reasonably" secure, but most likely not as secure as Apple's app sandboxing, and this should be borne in mind when assessing whether or not a PWA is appropriate, given the sensitivity, or not, of the data it will be handling, and the general security features of the devices where the app will be in use.
```

Comment:
Whilst this is all useful material, I think it is detracting from the core message about selecting the appropriate technology option.

I'd suggest putting this under a separate heading of "PWA standards".

Reply:
Not sure: the point of this material is to emphasise that PWA is a perfectly reasonable option, even on Apple devices (this is fast moving area so anyone who just Googles, or is just relying on what's in their noggin, will be retrieving a lot of out of date info that could create an unjustified prejudice against PWAs, especially on Apple. I know because I had to go through that learning process too).  That material certainly isn't an exhaustive list of PWA standards either.

Reply:
OK: now I've moved most of that material out of the standards into the guidance so it no longer overwhelms that section.